types/src/database_sql.rs

```diff
@@ -398,7 +396,76 @@ impl Database {
             "queries/update_word.sql",
             word.id,
             &source as _,
-            &commentary as _
+            &commentary as _,
+        )
+        .execute(&self.client)
+        .await?;
+
+        // If word segmentation was not changed, then return early since SQL update queries need to be called.
+        if word.segments.is_undefined() {
+            return Ok(word.id);
+        }
+
+        let document_id = query_file!("queries/word_by_id.sql", word.id)
+            .fetch_one(&self.client)
+            .await?
+            .document_id;
+
+        let segments = word.segments.take().unwrap();
+
+        let system_name: Option<CherokeeOrthography> = *(&segments[0].system.clone());
+
+        let (doc_id, gloss, word_id, index, morpheme, role): (
+            Vec<_>,
+            Vec<_>,
+            Vec<_>,
+            Vec<_>,
+            Vec<_>,
+            Vec<_>,
+        ) = segments
+            .into_iter()
+            .enumerate()
+            .map(move |(index, segment)| {
+                (
+                    document_id,
+                    segment.gloss,
+                    word.id,
+                    index as i64, // index of the segment in the word
+                    segment.morpheme,
+                    segment.role,
+                )
+            })
+            .multiunzip();
+
+        // First update any potentially new created local glosses.
+        query_file!(
+            "queries/upsert_local_morpheme_glosses.sql",
+            &*doc_id,
+            &*gloss
+        )
+        .execute(&self.client)
+        .await?;
+
+        // Then, convert the given glosses if they have an internal for to add to the database.
+        let internal_glosses: Vec<String> = query_file_scalar!(
+            "queries/find_internal_glosses.sql",
+            &*gloss,
+            match system_name {
```

Comment:
Whats up with this? You return TAOC no matter what, is the match here in case of supporting other systems in the future? Please add a comment here explaining if you think this makes sense. Also, if you expect to get only TAOC, then perhaps return an error in the else case instead of "TAOC".

Reply:
This is the main issue blocking my approval, because I want it to error out of we try to not use TAOC.

Reply:
Yeah, the match is there to support other systems in the future, but I was worried if the user somehow managed to get the backend call this function without "TAOC" enabled, then we should handle that case by having the system be TAOC be default since I didn't want there to be an error (in hindsight, makes less sense now I guess) but I can change this to return an error instead.